onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -893,6 +893,61 @@ Example 4:
           "Constrain to tensor(float).")
       .SetDoc(R"DOC(The WordConvEmbedding takes in a batch of sequence words and embed each word to a vector.)DOC");
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(ROIAlign)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Attr(
+          "spatial_scale",
+          "Multiplicative spatial scale factor to translate ROI coordinates "
+          "from their input spatial scale to the scale used when pooling, "
+          "i.e., spatial scale of the input feature map X relative to the "
+          "input image. E.g.; default is 1.0f. ",
+          AttributeProto::FLOAT,
+          1.f)
+      .Attr(
+          "pooled_h",
+          "default 1; Pooled output Y's height.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "pooled_w",
+          "default 1; Pooled output Y's width.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "sampling_ratio",
+          "Number of sampling points in the interpolation grid used to compute "
+          "the output value of each pooled output bin. If > 0, then exactly "
+          "sampling_ratio x sampling_ratio grid points are used. If == 0, then "
+          "an adaptive number of grid points are used (computed as "
+          "ceil(roi_width / pooled_w), and likewise for height). Default is 0.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .Attr(
+          "mode",
+          "The pooling method. Two modes are supported: 'avg' and 'max'. "
+          "Default is 'avg'.",
+          AttributeProto::STRING,
+          std::string("avg"))
+      .Input(0, "X", "Input data tensor from the previous operator; 4-D feature map of shape (N x C x H x W), where N is the batch size, C is the number of channels, and H and W are the height and the width of the data.", "T")
+      .Input(1, "rois", "RoIs (Regions of Interest2) to pool over; rois is 2-D input of shape (num_rois, 5) given as [[batch_id, x1, y1, x2, y2], ...]. The RoIs' coordinates are in the coordinate system of the input image.", "T")
+      .Output(0, "Y", "RoI pooled output, 4-D tesnor of shape (num_rois, C, pooled_h, pooled_w). The r-th batch element Y[r-1] is a pooled feature map corresponding to the r-th RoI X[r-1].", "T")
+      .TypeConstraint(
```

Comment:
>TypeConstraint [](start = 7, length = 14)

also add shape inference?

Reply:
Good point. I didn't add it because the facebook folks are already in the process of proposing this op and it'll be part of opset 9 and they'll be adding it as part of the PR.